server/app/src/main/java/com/ams/app/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,16 @@
+package com.ams.app.exception;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.server.ResponseStatusException;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+    @ExceptionHandler(ResponseStatusException.class)
+    public ResponseEntity<CustomErrorResponse> handleResponseStatusException(ResponseStatusException ex) {
+        CustomErrorResponse customErrorResponse = new CustomErrorResponse(ex.getRawStatusCode(), ex.getReason(), "/");
```

Comment:
why are we using ex.getReason(), instead of ex.getMessage()?

Reply:
In ResponseStatusException we can only pass reason as a parameter to display custom message. So used getReason()